lib/themes/dosomething/paraneue_dosomething/js/components/Slideshow.js

```diff
@@ -20,9 +20,12 @@ class Slideshow extends React.Component {
   }
 
   render() {
+    const user = this.props.user;
+    const campaign = this.props.campaign;
+
     return (
       <div className="slideshow">
-        {this.props.slides[this.state.activeSlide]}
+        {React.createElement(this.props.slides[this.state.activeSlide], {campaign: campaign, user: user})}
```

Comment:
I think you still want to create each of the slides with a unique key (like you had before, but then just grabbing the appropriate slide) so React knows about them all when transitioning things into and out of the DOM.